src/go/rpk/pkg/cli/cloud/mcp/mcp.go

```diff
@@ -209,3 +197,314 @@ func newStdioCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+		return []string{"claude", "claude-code"}, cobra.ShellCompDirectiveDefault
+	})
+	return cmd
+}
+
+// makePathPretty replaces home directory with ~/.
+func makePathPretty(path string) string {
+	if homeDir, err := os.UserHomeDir(); err == nil {
+		if strings.HasPrefix(path, homeDir) {
+			return "~" + strings.TrimPrefix(path, homeDir)
+		}
+	}
+	return path
+}
+
+// installMCPConfig installs MCP configuration for the specified client.
+func installMCPConfig(cfg *config.Config, mcpClient, serverName string, args []string) (string, error) {
+	var file string
+	switch mcpClient {
+	case "claude":
+		configDir, err := os.UserConfigDir()
+		if err != nil {
+			return "", fmt.Errorf("failed to get user configuration directory: %w", err)
+		}
+		file = filepath.Join(configDir, "Claude", "claude_desktop_config.json")
+	case "claude-code":
+		homeDir, err := os.UserHomeDir()
+		if err != nil {
+			return "", fmt.Errorf("failed to get user home directory: %w", err)
+		}
+		file = filepath.Join(homeDir, ".claude.json")
+	default:
+		return "", fmt.Errorf("unsupported client: %s", mcpClient)
+	}
+
+	fileMode := fz.FileMode(0o600)
+	jsonStr := `{}`
+	f, err := os.OpenFile(file, os.O_RDONLY, 0o600)
+	if err != nil && !os.IsNotExist(err) {
+		return "", fmt.Errorf("failed to open config file: %v", err)
+	}
+	defer f.Close()
+
+	if err == nil {
+		stat, err := f.Stat()
+		if err != nil {
+			return "", fmt.Errorf("failed to stat config file: %v", err)
+		}
+		fileMode = stat.Mode()
+
+		bytez, err := io.ReadAll(f)
+		if err != nil {
+			return "", fmt.Errorf("failed to read content of config file: %v", err)
+		}
+		jsonStr = string(bytez)
+	}
+
+	// Set command
+	serverKey := fmt.Sprintf("mcpServers.%s.command", serverName)
+	jsonStr, err = sjson.Set(jsonStr, serverKey, "rpk")
+	if err != nil {
+		return "", fmt.Errorf("failed to patch config: %v", err)
+	}
+
+	// Build base args with cloud environment if needed
+	var mcpArgs []string
+	if cfg.VirtualProfile().CloudEnvironment != "" {
+		mcpArgs = append(mcpArgs, "-X")
+		mcpArgs = append(mcpArgs, fmt.Sprintf("cloud_environment=%s", cfg.VirtualProfile().CloudEnvironment))
+	}
+
+	rpkConfigPath, err := config.DefaultRpkYamlPath()
+	if err != nil {
+		return "", fmt.Errorf("failed to get rpk config path: %w", err)
+	}
+	mcpArgs = append(mcpArgs, "--config", rpkConfigPath)
+	mcpArgs = append(mcpArgs, args...)
+
+	// Set args
+	argsKey := fmt.Sprintf("mcpServers.%s.args", serverName)
+	jsonStr, err = sjson.Set(jsonStr, argsKey, mcpArgs)
+	if err != nil {
+		return "", fmt.Errorf("failed to patch config: %v", err)
+	}
+
+	tmpFile := file + ".bak"
+	err = os.WriteFile(tmpFile, []byte(jsonStr), fileMode)
+	if err != nil {
+		return "", fmt.Errorf("failed to write file: %v", err)
+	}
+	err = os.Rename(file+".bak", file)
+	if err != nil {
+		return "", fmt.Errorf("failed to rename: %v", err)
+	}
```

Comment:
Check: https://github.com/redpanda-data/redpanda/blob/dev/src/go/rpk/pkg/os/file.go#L28 it might be just the same. We use it for atomic file writing for our config

Reply:
done